powerstore/resource_snapshotrule.go

```diff
@@ -0,0 +1,352 @@
+package powerstore
+
+import (
+	"context"
+	"log"
+	"strings"
+	"terraform-provider-powerstore/models"
+
+	"github.com/dell/gopowerstore"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type resourceSnapshotRuleType struct{}
+
+// GetSchema returns the schema for snapshotrule resource.
+func (r resourceSnapshotRuleType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:                types.StringType,
+				Computed:            true,
+				Description:         "The ID of the snapshot rule.",
+				MarkdownDescription: "The ID of the snapshot rule.",
+			},
+			"name": {
+				Type:                types.StringType,
+				Required:            true,
+				Description:         "The name of the snapshot rule.",
+				MarkdownDescription: "The name of the snapshot rule.",
+			},
+			"interval": {
+				Type:                types.StringType,
+				Optional:            true,
+				Computed:            true,
+				Description:         "The interval between snapshots taken by a snapshot rule, mutually exclusive with time_of_day parameter.",
+				MarkdownDescription: "The interval between snapshots taken by a snapshot rule.",
+			},
+			"time_of_day": {
+				Type:                types.StringType,
+				Optional:            true,
+				Computed:            true,
+				Description:         "The time of the day to take a daily snapshot, with format hh:mm, mutually exclusive with interval parameter.",
+				MarkdownDescription: "The time of the day to take a daily snapshot, with format hh:mm.",
+			},
+			"timezone": {
```

Comment:
Need to add default attribute for timezone. Default value is "UTC"

Reply:
this is by default, and server take care of it

Reply:
also, I cannot send it with every request as timezone can only be sent if time_of_day is provided and not other mutually exclusive interval param

Reply:
fixed